lib/taskana-core/src/test/java/acceptance/task/transfer/TransferTaskAccTest.java

```diff
@@ -435,4 +435,220 @@ void should_NotSetTheTransferFlagWithinBulkTransfer_When_SetTransferFlagNotReque
+  }
+
+  @WithAccessId(user = "teamlead-1", groups = GROUP_1_DN)
+  @Test
+  void should_TransferTaskAndSetOwner_When_WorkbasketKeyAndDomainIsProvided() throws Exception {
+    taskService.transferWithOwner(
+        "TKI:200000000000000000000000000000000066", "USER-1-2", "DOMAIN_A", "teamlead-1");
+
+    Task transferredTask = taskService.getTask("TKI:200000000000000000000000000000000066");
+    assertThat(transferredTask).isNotNull();
+    assertThat(transferredTask.isTransferred()).isTrue();
+    assertThat(transferredTask.isRead()).isFalse();
+    assertThat(transferredTask.getState()).isEqualTo(TaskState.READY);
+    assertThat(transferredTask.getOwner()).isEqualTo("teamlead-1");
+    assertThat(transferredTask.getWorkbasketSummary().getId())
+        .isEqualTo("WBI:100000000000000000000000000000000007");
+  }
+
+  @WithAccessId(user = "teamlead-1", groups = GROUP_1_DN)
+  @Test
+  void should_BulkTransferTasksAndSetOwner_When_WorkbasketKeyAndDomainIsProvided()
+      throws Exception {
+    final Instant before = Instant.now().truncatedTo(ChronoUnit.MILLIS);
+    List<String> taskIdList =
+        List.of(
+            "TKI:000000000000000000000000000000000008", "TKI:000000000000000000000000000000000009");
+
+    BulkOperationResults<String, TaskanaException> results =
+        taskService.transferTasksWithOwner("GPK_B_KSC_1", "DOMAIN_B", taskIdList, "teamlead-1");
+    assertThat(results.containsErrors()).isFalse();
+
+    final Workbasket wb =
+        taskanaEngine.getWorkbasketService().getWorkbasket("GPK_B_KSC_1", "DOMAIN_B");
+    Task transferredTask = taskService.getTask("TKI:000000000000000000000000000000000008");
+    assertThat(transferredTask).isNotNull();
+    assertThat(transferredTask.isTransferred()).isTrue();
+    assertThat(transferredTask.isRead()).isFalse();
+    assertThat(transferredTask.getState()).isEqualTo(TaskState.READY);
+    assertThat(transferredTask.getWorkbasketKey()).isEqualTo(wb.getKey());
+    assertThat(transferredTask.getDomain()).isEqualTo(wb.getDomain());
+    assertThat(transferredTask.getModified().isBefore(before)).isFalse();
+    assertThat(transferredTask.getOwner()).isEqualTo("teamlead-1");
+    transferredTask = taskService.getTask("TKI:000000000000000000000000000000000009");
+    assertThat(transferredTask).isNotNull();
+    assertThat(transferredTask.isTransferred()).isTrue();
+    assertThat(transferredTask.isRead()).isFalse();
+    assertThat(transferredTask.getState()).isEqualTo(TaskState.READY);
+    assertThat(transferredTask.getWorkbasketKey()).isEqualTo(wb.getKey());
+    assertThat(transferredTask.getDomain()).isEqualTo(wb.getDomain());
+    assertThat(transferredTask.getModified().isBefore(before)).isFalse();
+    assertThat(transferredTask.getOwner()).isEqualTo("teamlead-1");
+  }
+
+  @WithAccessId(user = "admin")
+  @Test
+  void should_SetTransferFlagToFalse_When_WithOwnerAndWorkbasketIdGiven() throws Exception {
```

Comment:
The test name seems confusing to me. What do you think about "should_SetTransferFlagAsSpecified_When_TransferringTaskWithOwnerUsingWorkbasketId"?

Reply:
Yes it sounds better